### PR TITLE
Oauth go dependency change

### PR DESCRIPTION
**Pull request by kdamichie**
This PR Addresses a needed field to be provided for OAuth security when using Microsoft Authenticator
<hr/>

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Ops

## What is the best way to verify this PR?

Standup a local instance of CIAB, and redirect your localhost to nightly URL.  Ensure your user credentials are in the DB.  Next log into TP using SSO option.  This should log you in without error.
## PR submission checklist
- [ ] This PR has tests 
- [x] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: docs/source/admin/traffic_ops.rst [hunk 1]
```diff
@@ -428,6 +428,7 @@ This file deals with the configuration parameters of running Traffic Ops itself.
...
 	:max_db_connections: An optional limit on the number of allowed concurrent connections to the Traffic Ops Database. If it is less than or equal to zero, there is no limit. Default if not specified is zero.
 	:oauth_client_secret: An optional secret string to be shared with OAuth-capable clients attempting to authenticate via OAuth. The default behavior if this is not defined - or is an empty string (``""``) or ``null`` is to disallow authentication via OAuth.
+	:oauth_user_attribute: An optional username string to be shared with OAuth-capable clients attempting to authenticate via OAuth. The default behavior if this is not defined - or is an empty string (``""``) or ``null`` is to disallow authentication via OAuth.
```
Comment (srijeet0406) [inline docs/source/admin/traffic_ops.rst:431]: The `-` should probably be in ````

File: traffic_ops/traffic_ops_golang/login/login.go [hunk 2]
```diff
@@ -483,17 +483,29 @@ func OauthLoginHandler(db *sqlx.DB, cfg config.Config) http.HandlerFunc {
...
-		if err != nil {
-			api.HandleErr(w, r, nil, http.StatusInternalServerError, nil, fmt.Errorf("Error decoding token with message: %w", err))
-			return
+		); err != nil {
+			if decodedToken, err = jwt.Parse(
+				[]byte(encodedToken),
+				jwt.WithVerifyAuto(false),
+				jwt.WithJWKSetFetcher(jwksFetcher),
+			); err != nil {
+				api.HandleErr(w, r, nil, http.StatusInternalServerError, nil, fmt.Errorf("Error decoding token with message: %w", err))
```
Comment (srijeet0406) [inline traffic_ops/traffic_ops_golang/login/login.go:497]: I know this is not your code, but while you're in here, mind fixing the error statement to start with a lower case letter?

File: dev/traffic_ops/cdn.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
 		"log_location_event": "stdout",
 		"whitelisted_oauth_urls": [],
 		"oauth_client_secret": "",
+		"oauth_user_attribute": "unique_name",
```
Comment (srijeet0406) [inline dev/traffic_ops/cdn.json:13]: Would these also need to be added to the ansible main.yml file?

Comment (kdamichie) [reply dev/traffic_ops/cdn.json:13]: @zrhoffman do we need it there?

Comment (zrhoffman) [reply dev/traffic_ops/cdn.json:13]: >> Would these also need to be added to the ansible main.yml file?
>>
> @zrhoffman do we need it there?

The Traffic Ops Ansible role doesn't yet have OAuth stuff on the ATC master branch. @dsouza93 is adding it in [`dsouza93:lab_oath](https://github.com/dsouza93/trafficcontrol/commits/lab_oath) (but has not PRed it yet).

So, adding it in #7619 is not really possible without creating conflicts.

**Review by srijeet0406 — APPROVED** [review]

**Review by kdamichie — COMMENTED** [review]
Looks good.  Tested, and working

**Pull request merged**